Dockerfile

```diff
@@ -3,29 +3,48 @@
 # This image includes support for Fortran and uses llvm-3.5 and gcc-4.7.
 #
 
-FROM ubuntu:14.04
+FROM ubuntu:16.04
 MAINTAINER Pablo Oliveira <pablo.oliveira@uvsq.fr>
 
+ENV LLVM_VERSION 3.9
+ENV GCC_VERSION 5
+ENV GCC_PATH /usr/lib/gcc/x86_64-linux-gnu/${GCC_VERSION}
+
 # Retrieve dependencies
-RUN apt-get -y update && apt-get -y install --no-install-recommends bash ca-certificates make git libmpfr-dev clang-3.5 llvm-3.5 llvm-3.5-dev dragonegg-4.7 gcc-4.7 g++-4.7 gfortran-4.7 libgfortran-4.7-dev autoconf automake libedit-dev libtool libz-dev python3 python3-numpy python3-matplotlib binutils vim && rm -rf /var/lib/apt/lists/
-ENV LIBRARY_PATH /usr/lib/gcc/x86_64-linux-gnu/4.7:/usr/lib/llvm-3.5/lib:$LIBRARY_PATH
+# RUN apt-get -y update && apt-get -y install --no-install-recommends software-properties-common
+# RUN add-apt-repository ppa:ubuntu-toolchain-r/test -y
+RUN apt-get -y update && apt-get -y install --no-install-recommends \
+    bash ca-certificates make git libmpfr-dev clang-${LLVM_VERSION} llvm-${LLVM_VERSION} llvm-${LLVM_VERSION}-dev \
+    gcc-${GCC_VERSION} gcc-${GCC_VERSION}-plugin-dev g++-${GCC_VERSION} gfortran-${GCC_VERSION} libgfortran-${GCC_VERSION}-dev \
+    autoconf automake libedit-dev libtool libz-dev \
+    python3 python3-numpy python3-matplotlib binutils vim && \
+    rm -rf /var/lib/apt/lists/
+
+ENV LIBRARY_PATH ${GCC_PATH}:$(llvm-config-${LLVM_VERSION} --prefix):$LIBRARY_PATH
+
+# Setup working directory
+VOLUME /workdir
+WORKDIR /workdir
+
+# Download dragonegg from a non official repo since the official version does not
+# work with llvm-3.6 and gcc-4.9
+RUN git clone -b gcc-5 https://github.com/tarunprabhu/DragonEgg.git
+WORKDIR DragonEgg
+RUN LLVM_CONFIG=llvm-config-${LLVM_VERSION} GCC=gcc-${GCC_VERSION} GCC=g++-${GCC_VERSION} make
+ENV DRAGONEGG_PATH=$PWD/dragonegg.so
+
+WORKDIR /workdir
 
 # Download and configure verificarlo from git master
 RUN \
   git clone https://github.com/verificarlo/verificarlo.git && \
   cd verificarlo && \
   ./autogen.sh && \
-  ./configure --with-llvm=/usr/lib/llvm-3.5/bin/ --with-dragonegg=/usr/lib/gcc/x86_64-linux-gnu/4.7/plugin/dragonegg.so CC=gcc-4.7 CXX=g++-4.7
+  ./configure --with-llvm=$(llvm-config-${LLVM_VERSION} --prefix) --with-dragonegg=${DRAGONEGG_PATH} CC=${GCC_VERSION} CXX=g++-{GCC_VERSION}
 
 # Build and test verificarlo
 
 ENV LD_LIBRARY_PATH /usr/local/lib:$LD_LIBRARY_PATH
 ENV PATH /usr/local/bin:$PATH
 ENV PYTHONPATH /usr/local/lib/python3.4/site-packages/:$PYTHONPATH
 RUN cd verificarlo && make && make install && make installcheck
-
-# Setup working directory
```

Comment:
Why should we remove these ? Is the tutorial still working with this image, if we make this change ?

Reply:
It was a mistake. Thank you for pointing out.